src/routes/api/tasks/index.ts

```diff
@@ -299,6 +299,50 @@ const plugin: FastifyPluginAsyncTypebox = async (fastify) => {
       )
     }
   )
+
+  fastify.delete(
+    '/:filename/image',
+    {
+      schema: {
+        params: Type.Object({
+          filename: Type.String()
+        }),
+        response: {
+          204: Type.Null(),
+          404: Type.Object({ message: Type.String() })
+        },
+        tags: ['Tasks']
+      }
+    },
+    async function (request, reply) {
+      const { filename } = request.params
+
+      return fastify.knex.transaction(async (trx) => {
+        const task = await trx<Task>('tasks').select('filename').where({ filename }).first()
+        if (!task) {
+          return reply.notFound(`No task has filename "${filename}"`)
+        }
```

Comment:
Maybe you can try to update, and if it return `0`, it means no row where updated, so you can guess that the task was not found. Maybe we can apply the same strategy for upload endpoint btw.

The reason is to make one query instead of 2.

Reply:
Ok, should i make the change for the upload endpoint also or i can do it in another pr / issue ?

Reply:
Yep, I think you can do it on this PR if you want.
You put the logic after file size/mime validation.